### DISCUSSION TITLE
How to install without LSP Zero on Neovim

### DISCUSSION THREAD

**Discussion by uhooi** [supabase-community/postgres-language-server#24] [General]
# Improve documentation

## Link 

https://github.com/supabase/postgres_lsp/blob/d4e971eb4e526e568d44a17fc878807069d515ea/README.md#neovim

## Describe the problem

Postgres Language Server installation appears to be dependent on [LSP Zero](https://github.com/VonHeikemen/lsp-zero.nvim).
I would like to know how to install it without dependencies.

## Describe the improvement

Avoid reliance on LSP Zero.

## Additional context

**Comment by psteinroe** [comment]
hey! thanks for the issue. are you interested in collaborating? right now, the language server is not ready for use, and installing the server is only recommended for debugging purposed when working on the project. I will make sure to put the server into all the registries once it's ready to simplify the installation.

**Comment by 9mm** [comment]
@psteinroe probably not hot take but you got a bunch of interest from HN, that's how I found this. Maybe good to ride that wave instead of everyone coming here and now everything is just "it's not done." I'm also wondering this basically (i use lsp-zero) and if there's no docs or anything how to add I'll likely forget about it and move on (I did write a note to myself to remember, but its in a long list of notes 😬)

**Comment by psteinroe** [comment]
@9mm I fully understand your concerns. To be transparent, this kinda blew up in my hands (and still is). We tried to make it clear in both the HN post and in the Readme that this project is in a very early state, and we want to develop it in the open. It makes me very happy that so many people share my need of a language server for postgres, but that unfortunately does not change the fact that it's still early in development. If you open any sql file other than the example, the server will immediately crash. There is literally zero value in it as of now. I will publish it to all registries and update the readme once it's at least in a usable state, but it will take a while. I am sincerely sorry if that disappoints you.

**Comment by 9mm** [comment]
@psteinroe I'm not disappointed at all, and not trying to rush you. Just giving you my unsolicited .02 if it makes a difference about the swaths of coomers descending upon your repo wanting things, and the next minute, they disappear

**Comment by kiwicopple** [comment]
hey @9mm - just to be clear, we posted this as a "Show HN" with a lot of clear caveats that it wasn't ready:

> _"this is still in early development, and mostly useful for testers/collaborators. the majority of work is still ahead, but we've verified that the approach works. we're making it public now so that we can develop it in the open with input from the community."_

that comment still stands. We still have to make it work for general use, and hopefully a few more contributors saw the post and want to help out.

> swaths of coomers descending upon your repo wanting things, and the next minute, they disappear

This is quite honestly fine, we can't rush a good solution. We can post this again when it's "production ready", and perhaps they will see it again. 

For now, I'll convert this one into a Discussion, but we'll keep it open for any more comments and updates. Let's use the issues to track bugs wherever possible 🙏

**Comment by rurumimic** [comment]
Hi. I'm also making a SQL LSP recently. I'm glad to see this repo.
I don't use lsp-zero, but so far it seems to work well on both nvim and vscode.
Great!

![postgres_lsp](https://github.com/supabase/postgres_lsp/assets/10915235/c859e8b6-5ffc-4a55-b91e-64649f86baf4)
